src/Symfony/Component/BrowserKit/Client.php

```diff
@@ -563,6 +571,23 @@ public function followRedirect()
         return $response;
     }
 
+    /**
+     * Determines if the response has a meta refresh.
+     *
+     * @return string|null Redirect URI if response has a meta refresh
+     */
+    private function getMetaRefresh(): ?string
+    {
+        $metaRefresh = $this->getCrawler()->filter('head meta[http-equiv="refresh"]');
+        foreach ($metaRefresh->extract(array('content')) as $content) {
+            if (preg_match('/.*URL=(.*)$/i', $content, $m)) {
```

Comment:
about the regexp, what about restricting to zero timeouts? and be more strict about it, eg:
```php
if (preg_match('/^\s*0\s*;\s*URL\s*=\s*(?|'([^']++)|"([^"]++)|([^'"].*))/i', $content, $m)) {
    return str_replace("\t\r\n", '', rtrim($m[1]);
}
```

parsing logic extracted from https://dev.w3.org/html5/spec-preview/the-meta-element.html#attr-meta-http-equiv-refresh

Reply:
Thanks for this regex! It helped out a lot :)